src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -457,7 +475,15 @@ private function validateClassNode($object, $cacheKey, ClassMetadataInterface $m
             $defaultOverridden = false;
 
             // Use the object hash for group sequences
-            $groupHash = \is_object($group) ? spl_object_hash($group) : $group;
```

Comment:
would replacing this line by `$groupHash = serialize($group);` fix the issue?

Reply:
It works as long as one does not add constraints to class metadata between 2 validations on the same group. I guess it's hacky enough to be ignored.

Reply:
I added a test for that btw.